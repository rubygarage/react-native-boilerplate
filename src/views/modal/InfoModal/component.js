import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import ModalView from 'views/shared/ModalView';
import ModalHeader from 'views/shared/ModalView/ModalHeader';
import ModalBody from 'views/shared/ModalView/ModalBody';
import ModalFooter from 'views/shared/ModalView/ModalFooter';
import ModalText from 'views/shared/ModalView/ModalText';
import Button from 'views/shared/Button';

const InfoModal = ({
  isVisible, onCloseModal, onDestroyModal, title, detail,
}) => (
  <ModalView
    isVisible={isVisible}
    onModalHide={onDestroyModal}
  >
    <ModalHeader>
      {title}
    </ModalHeader>
    <ModalBody>
      <ModalText>
        {detail}
      </ModalText>
    </ModalBody>
    <ModalFooter>
      <Button onPress={onCloseModal}>
        <FormattedMessage id="shared.ok" />
      </Button>
    </ModalFooter>
  </ModalView>
);

InfoModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onCloseModal: PropTypes.func.isRequired,
  onDestroyModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  detail: PropTypes.string.isRequired,
};

export default InfoModal;
