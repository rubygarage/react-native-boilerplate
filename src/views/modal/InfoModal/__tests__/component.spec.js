import React from 'react';
import { shallow } from 'enzyme';

import InfoModal from '../component';

describe('InfoModal component', () => {
  const defaultProps = {
    isVisible: true,
    onCloseModal: jest.fn(),
    onDestroyModal: jest.fn(),
    title: 'test title',
    detail: 'test detail',
  };

  const component = shallow(<InfoModal {...defaultProps} />);

  it('renders correctly default', () => {
    expect(component).toMatchSnapshot();
  });
});
