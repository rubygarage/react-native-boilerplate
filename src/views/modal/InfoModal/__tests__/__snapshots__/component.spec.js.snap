// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal component renders correctly default 1`] = `
<ModalView
  isVisible={true}
  onModalHide={[MockFunction]}
>
  <WithTheme(ModalHeader)>
    test title
  </WithTheme(ModalHeader)>
  <ModalBody>
    <Styled(Text)>
      test detail
    </Styled(Text)>
  </ModalBody>
  <ModalFooter>
    <WithTheme(Button)
      onPress={[MockFunction]}
    >
      <FormattedMessage
        id="shared.ok"
      />
    </WithTheme(Button)>
  </ModalFooter>
</ModalView>
`;
